assembly/pipeline.py

```diff
@@ -52,17 +52,22 @@ def httpGetFile(url, filepath=""):
         urllib.request.urlretrieve(url, filepath)
         return True
 
-def gunzip(inputpath="", outputpath=""):
-    if (outputpath == ""):
-        with gzip.open(inputpath, 'rb') as f:
-            gzContent = f.read()
-        return gzContent
-    else:
-        with gzip.open(inputpath, 'rb') as f:
-            gzContent = f.read()
-        with open(outputpath, 'wb') as out:
-            out.write(gzContent)
+def gunzip(file_path):
+    with gzip.open(file_path, 'rb') as f:
+        gzContent = f.read()
+    with open(file_path.replace(".gz",""), 'wb') as out:
+        out.write(gzContent)    
+    os.remove(file_path)
+    if (os.path.exists(file_path.replace(".gz",""))):
         return True
+    else:
+        raise Exception("Reference genome downloaded, but cannot be gunzipped")
+
+def check_file_exist_and_parse(path, parser):
```

Comment:
What happens if we try to run a parser directly on a file that doesn't exist. Doesn't that throw an Exception already?